### ISSUE TITLE
Remove duplicate crate dependencies (e.g. multiple versions of nix)

### ISSUE THREAD

**Issue by lsm5** [containers/netavark#204]
v1.0.0-rc1 depends on three versions of the nix crate. I see that they are pulled in because of iptables, rtnetlink and zbus each require a different version of the crate. But is there any cargo magic to only pull in 1 (latest | compatible) version ?

**Comment by Luap99** [comment]
My google search doesn't seem to find a solution for that other than patching the dependencies :/

**Comment by mskarbek** [comment]
Nope, there is no solution. That's what you get when you drop global dependency resolution. Each crate can depend on any version they want locally, and `cargo` will download and build that for them. If you want single nix you would have to either approach each crate maintainer and ask them to update their dependencies or, as @Luap99 mentioned, patch them yourself in a fork.
